node/core/candidate-validation/src/lib.rs

```diff
@@ -347,16 +347,16 @@ async fn spawn_validate_exhaustive(
 /// are passed, `Err` otherwise.
 fn perform_basic_checks(
 	candidate: &CandidateDescriptor,
-	max_block_data_size: Option<u64>,
+	max_pov_size: u32,
 	pov: &PoV,
 ) -> Result<(), InvalidCandidate> {
+	use std::convert::TryInto;
+
 	let encoded_pov = pov.encode();
 	let hash = pov.hash();
 
-	if let Some(max_size) = max_block_data_size {
-		if encoded_pov.len() as u64 > max_size {
-			return Err(InvalidCandidate::ParamsTooLarge(encoded_pov.len() as u64));
-		}
+	if encoded_pov.len().try_into().unwrap_or(!0) > max_pov_size {
```

Comment:
Either switch the sides or just cast the max pov size to usize as this fits into usize. Than you don't need TryInto